etna/records/templatetags/records_tags.py

```diff
@@ -10,20 +11,25 @@
 
 
 @register.simple_tag
-def record_url(record: Union[Record, Dict[str, Any]]) -> str:
+def record_url(
+    record: Union[Record, Dict[str, Any]], is_editorial: bool = False
+) -> str:
     """
     Return the URL for the provided `record` dict; which could either be a
     full/transformed result from the fetch() endpoint, OR a raw result from
     the search() endpoint.
     """
+    iaid = record.get("iaid")
+    if iaid and is_editorial and settings.FEATURE_RECORD_LINKS_GO_TO_DISCOVERY:
+        return f"https://discovery.nationalarchives.gov.uk/details/r/{iaid}"
     if ref := record.get("reference_number", record.get("referenceNumber")):
```

Comment:
`reference_number` is will be used for Insights, Can you please give an example to test `referenceNumber` ?

Reply:
Can we please stick to reviewing what's in-scope for this PR?

Reply:
Thanks for our discussion. The example to test is for Catalogue Search Results call with `referenceNumber`.